### PR TITLE
SSH retry

### PR DESCRIPTION
**Pull request by mdarby**
Customers have asked for a way to retry the SSH portion of creating a server. Basically we don't know what they're installing/waiting for on the server so SSH may or may not wait long enough and otherwise err out. This patch introduces `ssh_retry_every` option as a timer for a retry loop.

### CODE REVIEW

**Comment by martinb3** [comment]
This seems like a useful feature. I would avoid re-purposing a configuration option that already exists, and might be in use, as I feel would we need to bump major version. I'd rather see this be a new option, so we could only bump minor. The reason I think there should be an option is that I could see cases where people do or don't want this behavior, and might feel pretty strongly about it.

**Comment by mdarby** [comment]
Great call. I'll add a new config option.

File: lib/chef/knife/rackspace_server_create.rb [hunk 1]
```diff
@@ -459,7 +467,7 @@ def run
       else
         print "\n#{ui.color("Waiting for sshd", :magenta)}"
         print(".") until tcp_test_ssh(bootstrap_ip_address) {
-          sleep @initial_sleep_delay ||= 10
+          # sleep @initial_sleep_delay ||= 10
```
Comment (martinb3) [inline lib/chef/knife/rackspace_server_create.rb]: If we don't remove this, and leave it commented, could we leave a comment as a breadcrumb for why? Just to provide later clue when someone comes across this and wonders "Why did they do this?"?

**Comment by martinb3** [comment]
Everything else looks good to me. :)

**Comment by mdarby** [comment]
Sweet. This is most def a WIP (I will mark as so) -- hoping for finalization today.

**Comment by mdarby** [comment]
OK, I went a smarter route with the retry loop -- to test just run with --retry-ssh-every <seconds>

**Comment by mdarby** [comment]
@martinb3 I think this is reviewable now when you have a moment.

**Comment by martinb3** [comment]
@mdarby For folks who want the old functionality, is there a way to make this not go on forever, by default? Say like try 5 loops and bail? Just so it doesn't 'never return' -- that would also feel like a breaking change if it became an infinite loop by default, when a server is unavailable. Everything else looks good -- sorry to be so picky!

**Comment by mdarby** [comment]
no no no, please be picky. I'll add another option that limits the number of retries. Thank for your help!

**Comment by mdarby** [comment]
OK, I've added a new option `--retry-ssh-limit` and it defaults to 5.

**Comment by martinb3** [comment]
:+1: Looks good to me. Ready to merge?

**Comment by mdarby** [comment]
Ready if you are!

**Comment by martinb3** [comment]
:tada:

**Comment by mdarby** [comment]
When do you think we can cut a gem @jjasghar? I have a customer that's quite hot on this...

**Comment by jjasghar** [comment]
I think if we get this: https://github.com/chef/knife-rackspace/pull/108 merged ;) then it'll be up and running

**Pull request merged**